src/v/utils/tests/process_spawn_tests.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "test_utils/async.h"
+#include "utils/process.h"
+#include "vassert.h"
+
+#include <seastar/testing/thread_test_case.hh>
+
+#include <chrono>
+
+inline ss::logger test_log("test");
+
+std::filesystem::path get_trap_script() {
+    auto trap_script = std::getenv("HANDLE_SIGTERM_SCRIPT");
+    vassert(trap_script, "$HANDLE_SIGTERM_SCRIPT does not exist");
+    BOOST_REQUIRE_MESSAGE(
+      ss::file_exists(trap_script).get0(),
+      std::string(trap_script) + " does not exist");
+    return std::filesystem::path{trap_script};
+}
+
+SEASTAR_THREAD_TEST_CASE(process_start) {
+    std::filesystem::path cmd{"/usr/bin/sleep"};
```

Comment:
I would recommend:

Not using `sleep` as it could race, even if 2 seconds does make it fairly unlikely.